langchain/llms/mlflow_gateway.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+from typing import Any, List, Dict, Mapping, Optional
+from langchain.callbacks.manager import CallbackManagerForLLMRun
+from langchain.llms.base import LLM
+from pydantic import BaseModel, Extra
+
+
+class Params(BaseModel, extra=Extra.allow):
+    temperature: float = 0.0
+    stop: List[str] | None = None
+    max_tokens: Optional[int] = None
+    candidate_count: int = 5
+
+
+class MlflowGateway(LLM):
+    route: str
+    gateway_uri: str | None = None
+    params: Params | None = None
+
+    def __init__(self, **kwargs: Any):
+        import mlflow.gateway
+
+        super().__init__(**kwargs)
+        if self.gateway_uri:
+            mlflow_gateway.set_gateway_uri(self.gateway_uri)
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        return {
+            "gateway_uri": self.gateway_uri,
+            "route": self.route,
+            "params": self.params.dict(),
+        }
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        return self._default_params
+
+    def _call(
+        self,
+        prompt: str,
+        stop: List[str] | None = None,
+        run_manager: CallbackManagerForLLMRun | None = None,
+        **kwargs: Any,
+    ) -> str:
+        import mlflow.gateway
+
+        data = {
+            "prompt": prompt,
+            **self.params.dict(exclude_none=True),
+        }
+        if s := (stop or self.params.stop):
+            data["stop"] = s
+        resp = mlflow.gateway.query(self.route, data=data)
+        return resp["candidates"][0]["text"]
+
+    @property
+    def _llm_type(self) -> str:
+        return "mlflow-gateway"
```

Comment:
@dbczumar @BenWilson2

```suggestion
        return "mlflow-ai-gateway"
```

should we use `mlflow-ai-gateway`?

Reply:
+1 to just match it to what people will see in docs investigations.

Reply:
made the change!